matplotlib-chord.py

```diff
@@ -162,13 +164,14 @@ def chordDiagram(X, ax, colors=None, width=0.1, pad=2, chordwidth=0.7):
     x = X.sum(axis = 1) # sum over rows
     ax.set_xlim(-1.1, 1.1)
     ax.set_ylim(-1.1, 1.1)
-
-    if colors is None:
-    # use d3.js category10 https://github.com/d3/d3-3.x-api-reference/blob/master/Ordinal-Scales.md#category10
+    
+    import matplotlib.pyplot as plt
+    if (len(x) > 10) or (colors is None):
+        cm = plt.get_cmap(cmap) 
+        colors = cm(np.linspace(0,1,len(x)))[:,:3]
+    else:
         colors = ['#1f77b4', '#ff7f0e', '#2ca02c', '#d62728', '#9467bd',
```

Comment:
The d3.js category10 colors have been used as the default colors for matplotlib v2.x. It has a better visualization than sequential color map to represent category data.

Reply:
Ok, is there any colormap in matplotlib including the d3.js category10? I couldn't find them..

Reply:
There is no color map for d3.js category10 in matplotlib. They are used as the default color cycles for `plt.plot()`.  Please check this: https://matplotlib.org/users/dflt_style_changes.html#colors-color-cycles-and-color-maps

Usually colormap is used for continuous variables, rather than category data. Therefore, my opinion is to use category10 to represent the data if there are less than 10 categories.

Reply:
Perfect then, we'll stick with category10 if less than 10 categories and no colormap specified.